src/utils.ts

```diff
@@ -17,6 +19,39 @@ export const sha256Multihash = (str: string): string => {
   return Multihash.encode(Buffer.from(sha256(str)), 'sha2-256').toString('hex')
 }
 
+const multicodecPubkeyTable = {
+  'secp256k1': 0xe7,
+  'x25519': 0xec,
+  'ed25519': 0xed
+}
+
+function encodeKey(key: Uint8Array, keyType: string) {
+  const buf = new Uint8Array(key.length + 2)
+  if (!multicodecPubkeyTable[keyType]) {
+    throw new Error(`Key type "${keyType}" not supported.`)
+  }
+  buf[0] = multicodecPubkeyTable[keyType]
+  // The multicodec is encoded as a varint so we need to add this.
+  // See js-multicodec for a general implementation
+  buf[1] = 0x01
+  buf.set(key, 2)
+  return `z${bs58.encode(buf)}`
+}
+
+export const gen3IDgenesis = (pubkeys: any): any => {
```

Comment:
Maybe we could allow extra keys to be provided here so we could add the `idx` field?

Reply:
`idx` can not be part of the genesis record. Will add the idx property and documents in the upcoming PR :)